android/src/app/organicmaps/car/SurfaceRenderer.java

```diff
@@ -0,0 +1,175 @@
+package app.organicmaps.car;
+
+import android.graphics.Rect;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.car.app.AppManager;
+import androidx.car.app.CarContext;
+import androidx.car.app.CarToast;
+import androidx.car.app.SurfaceCallback;
+import androidx.car.app.SurfaceContainer;
+import androidx.lifecycle.DefaultLifecycleObserver;
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleOwner;
+
+import app.organicmaps.Map;
+import app.organicmaps.R;
+
+public class SurfaceRenderer implements DefaultLifecycleObserver, SurfaceCallback
+{
+  private static final String TAG = SurfaceRenderer.class.getSimpleName();
+
+  private final CarContext mCarContext;
+  private final Map mMap;
+
+  @Nullable
+  private Rect mVisibleArea;
+  @Nullable
+  private Rect mStableArea;
+
+  public SurfaceRenderer(@NonNull CarContext carContext, @NonNull Lifecycle lifecycle)
+  {
+    Log.d(TAG, "SurfaceRenderer()");
+    mCarContext = carContext;
+    mMap = new Map();
+    lifecycle.addObserver(this);
+  }
+
+  @Override
+  public void onSurfaceAvailable(@NonNull SurfaceContainer surfaceContainer)
+  {
+    Log.d(TAG, "Surface available " + surfaceContainer);
+    mMap.onSurfaceCreated(
```

Comment:
Could you please explain how both mobile and car work at the same time? On my phone I still see the map, but with different DPI. I don't really understand how is it possible. 

https://github.com/organicmaps/organicmaps/blob/e9056f11fbc2cf3c41850701302bba305cbde95a/android/src/app/organicmaps/Map.java#L120-L123

Reply:
Никак. Мы это обсуждали еще в первых сообщениях этого PR.

> @biodranik, please don't overcomplicate things. Yes, it is technically possible in theory. Duplicating in OpenGL/Vulkan should be relatively easy, but the primary challenge on this road will be different screen sizes/resolutions in Auto and on the device. Basically, it will require the full duplication of rendering logic. As far as I can see, entire Google Inc. and Yandex were unable to implement this feature in their Android Auto implementations so far. Let's get Auto screen working first with disabled rendering in the main app.

Поддержки одновременной работы на телефоне и на AA не будет. 
Подробнее здесь https://github.com/organicmaps/organicmaps/pull/3806#discussion_r1043697281

Reply:
Нужен удобный UX. Вариант, когда например нужно выбрать сначала маршрут на телефоне, и только потом заработает карта навигатора AA, тоже можно рассмотреть. А вообще, лучше сделать невозможное. По возможности.